front-end/app/service/basket.service.ts

```diff
@@ -1,6 +1,54 @@
 import { Injectable } from '@angular/core';
 import { BasketService as SharedBasketService } from '../../../shared/services/basket-service';
 
+const STORAGE_KEY = 'basket';
+
 @Injectable()
 export class BasketService extends SharedBasketService {
+    constructor() {
+        super();
+        this.loadItems();
+    }
+
+    addToBasket(item: BasketItem): void {
+        super.addToBasket(item);
+        this.saveItems();
+    }
+
+    increase(item: OrderLineItem) {
+        super.increase(item);
+        this.saveItems();
+    }
+
+    decrease(item: OrderLineItem) {
+        super.decrease(item);
+        this.saveItems();
+    }
+
+    removeItem(item: OrderLineItem): void {
+        super.removeItem(item);
+        this.saveItems();
+    }
+
+    removeAll(): void {
+        super.removeAll();
+        this.saveItems();
+        localStorage.removeItem(STORAGE_KEY);
```

Comment:
You don't need this line of code. The storage will simply have an empty basket which is perfectly fine.